### ISSUE TITLE
Extend footprint_filter for non circular robots

### ISSUE THREAD

**Issue by jcarlosgm30** [ros-perception/laser_filters#193]
Hi!

The current implementation of footprint_filter is only for circular robots. It would be very useful if we extend the filter to be used by non-circular robots including a footprint in a similar format as NAV2.

what do you think about this? @jonbinney

**Comment by jonbinney** [comment]
Yes, definitely useful! I think the [polygon filter](https://github.com/ros-perception/laser_filters/blob/noetic-devel/include/laser_filters/polygon_filter.h) does that in ROS1, and there is actually a PR by @berend-kupers to port it (and other filters) to ROS2: https://github.com/ros-perception/laser_filters/pull/189/files . Its a big PR and I haven't had time to review it - if you have some time and can do a review that would help move it along!

**Comment by AntoBrandi** [comment]
I see that https://github.com/ros-perception/laser_filters/pull/189 has been merged 🎉
Will it be released to humble?

**Comment by jonbinney** [comment]
Yes, I'll do a release once I get https://github.com/ros-perception/laser_filters/pull/209 and https://github.com/ros-perception/laser_filters/pull/206 merged; should be this week.
